### PR TITLE
[knapsack] Updates subfolder syntax and adds separate API key for engine tests

### PR DESCRIPTION
**Pull request by filipefurtad0**
#### What? Why?

Follow-up from #9895 

**tl;dr**

This PR:
- adds a separate key for the CI nodes running the `knapsack_rspec_engines` job. 
- explicitly includes subfolders to comply to Dir.glob syntax (details [here](https://knapsackpro.com/faq/question/dir-glob-pattern-examples-for-knapsack_pro_test_file_pattern-and-knapsack_pro_test_file_exclude_pattern))

----

We need a separate KS-API key for each job to run correctly and so that the metrics can be viewed on the KS-dashboard. While addressing this, I noticed that the `build.yml` file was using `Dir.glob` syntax in some cases, and "rspec syntax" in others. For example, running:

`bundle exec rspec spec/system`

 runs all `*_spec.rb` files, subfolders included. However, in `build.yml` Not explicitly including subfolders in the folder pattern leads to missing out some specs. To illustrate this:

Before this PR
```
irb(main):007:0> Dir.glob("{spec/system/**/*_spec.rb}").count
=> 99
```
After this PR
```
irb(main):008:0> Dir.glob("{spec/system/**{,/*/**}/*_spec.rb}").count
=> 197
```

#### What should we test?

Green build.

#### Release notes

Changelog Category: Technical changes

The title of the pull request will be included in the release notes.

### CODE REVIEW

**Review by dacook — APPROVED** [review]
I have to confess I tried to understand how this works, but don't!

I thought the 'glob' pattern `**` should match zero or more directories, and recurse indefinitely. But I can see it only recurses one level, hence why you've added brace expansion here for the next level. 

If there's ever another level deeper, those specs would get ignored and we might not notice.. (like already happened ;) ) Is there any other ways to do it, maybe with a regex filter instead?  This seems like a common requirement, so surely someone has come up with a solution?

In the short term though, we are missing specs, so we'd better apply this fix quick.

**Comment by filipefurtad0** [comment]
> I have to confess I tried to understand how this works, but don't!

I have to be totally honest: I don't follow it as well. The change was proposed by following the suggestion in the documentation. I'm still looking at this in more detail, as I'm not 100% sure it is actually doing what we need.

More on this as soon as I have an answer.

**Comment by dacook** [comment]
Thanks for looking into this Filipe. So to be sure I understand, in summary:

* The extra `{,/*/**}` turned out not to be necessary?
* But it is consistent with the documentation so you think it's good to keep it?

Personally I think it's better not to have the extra code if it's not needed, and would suggest removing it if possible, to avoid future confusion.

**Comment by filipefurtad0** [comment]
> Personally I think it's better not to have the extra code if it's not needed, and would suggest removing it if possible, to avoid future confusion.

Indeed, that's a good idea :+1:

I'll remove the unnecessary code and make sure this has the expected result by looking at the knapsack-dashboard on the respective build runs. This should be the [PR (still a draft)](https://github.com/openfoodfoundation/openfoodnetwork/pull/10071). Thanks!
